src/lib/HST/Feature/GuardElimination.hs

```diff
@@ -194,14 +197,14 @@ applyGEMatches ms | any hasGuards ms = return <$> applyGE ms
 -- | Applies guard elimination to the rules of a function declaration.
 applyGE :: Member Fresh r => [S.Match a] -> Sem r (S.Match a)
 applyGE ms = do
-  let name  = getMatchName (head ms)
-      gexps = map matchToGExp ms
-      arity = length (gExpPats (head gexps))
+  let name    = getMatchName (head ms)
+      lhsSpan = S.getSrcSpan (head ms)
+      gexps   = map matchToGExp ms
+      arity   = length (gExpPats (head gexps))
   varPats <- replicateM arity (freshVarPat genericFreshPrefix)
   expr' <- generateLet (map S.patToExp varPats) defaultErrorExp gexps
   return
-    $ S.Match S.NoSrcSpan name varPats (S.UnGuardedRhs S.NoSrcSpan expr')
-    Nothing
+    $ S.Match lhsSpan name varPats (S.UnGuardedRhs S.NoSrcSpan expr') Nothing
```

Comment:
Just as in the `useAlgo` function in the `Application` module, you could pass the source span of the `FunBind` to this function and use it here for the `Match` and the `UnGuardedRhs`.